lib/Guacamole.pm

```diff
@@ -1376,11 +1376,16 @@ VersionNumberSegment ~ [0-9] [0-9] [0-9]
                      | [0-9] [0-9]
                      | [0-9]
 
+HexPrefix ~ '0x'
+HexDigits ~ [0-9a-fA-F]+
+HexNumber ~ HexPrefix HexDigits
+
 LitNumber ::= LitNumberDec
 LitNumberDec ~ Digits Period Digits
              | Digits Period
              | Period Digits
              | Digits
+             | HexNumber
```

Comment:
Hexademical numbers should have their own distinct AST node instead of being lumped together with decimals.

Looks like octal support is missing too (worse, they are mis-parsed as decimals).

And maybe we should support `_` as a digit separator?

Reply:
Yes, there should be separate productions for decimal, hexadecimal, octal (if you want, I don't really care for those) and binary (these are far more useful).

Also, here is my vote to support `_` as a digit separator: +1.